packages/bot-api/src/main/Bot.ts

```diff
@@ -0,0 +1,123 @@
+
+const logdown = require('logdown');
+
+class Bot {
+  public account: Account | undefined;
+
+  private readonly credentials: {email: string; password: string};
+  private readonly config: BotConfig;
+  private readonly handlers: Map<string, MessageHandler>;
+  private readonly logger: any = logdown('@wireapp/standup-bot/StandupBot', {
+    logger: console,
+    markdown: false,
+  });
+
+  constructor(credentials: {email: string; password: string}, config: BotConfig = {conversations: [], owners: []}) {
+    this.config = config;
+    this.credentials = credentials;
+    this.handlers = new Map();
+  }
+
+  public addHandler(handler: MessageHandler) {
+    this.handlers.set(new UUID(4).format(), handler);
+  }
+
+  public removeHandler(key: string) {
+    this.handlers.delete(key);
+  }
+
+  private isAllowedConversation(conversationId: string): boolean {
+    return this.config.conversations.length === 0 ? true : this.config.conversations.includes(conversationId);
+  }
+
+  private isOwner(userId: string): boolean {
+    return this.config.owners.length === 0 ? true : this.config.owners.includes(userId);
+  }
+
+  public async start(): Promise<boolean> {
+    const login = {
+      clientType: ClientType.TEMPORARY,
+      email: this.credentials.email,
+      password: this.credentials.password,
+    };
+    const backend = APIClient.BACKEND.PRODUCTION;
+    const engine = new MemoryEngine();
+    await engine.init(this.credentials.email);
+    const apiClient = new APIClient(new Config(engine, backend));
+    this.account = new Account(apiClient);
+    this.account.on(PayloadBundleType.TEXT, async (payload: PayloadBundleIncoming) => {
+      const conversationId = String(payload.conversation);
+      if (this.validateMessage(conversationId, payload.from)) {
+        this.logger.info('Processing message ...');
+        try {
+          this.handlers.forEach(handler => handler.handleEvent(payload));
+        } catch (error) {
+          this.logger.error(`An error occurred during text handling: ${error.message}`, error);
+        }
+      }
+    });
+    this.account.on(PayloadBundleType.CONNECTION_REQUEST, async (payload: PayloadBundleIncoming) => {
+      const connection: Connection = payload.content as Connection;
+      if (connection.status === ConnectionStatus.PENDING && connection.conversation) {
+        const {conversation, to: userId} = connection;
+        if (this.validateMessage(String(conversation), userId)) {
+          this.logger.info('Processing connection request ...');
+          try {
+            this.handlers.forEach(handler => handler.handleEvent(payload));
+          } catch (error) {
+            this.logger.error(`An error occurred during connection request handling: ${error.message}`, error);
+          }
+        }
+      }
+    });
+    await this.account.login(login);
+    await this.account.listen();
+    this.handlers.forEach(handler => (handler.account = this.account));
+    return true;
+  }
```

Comment:
This means the bot API can handle only `PayloadBundleType.TEXT` and `PayloadBundleType.CONNECTION_REQUEST`?

Reply:
You are right, we should think of a generic way to get all the events possible.